Drv/Ip/IpSocket.cpp

```diff
@@ -160,8 +168,11 @@ SocketIpStatus IpSocket::open() {
 SocketIpStatus IpSocket::send(const U8* const data, const U32 size) {
     U32 total = 0;
     I32 sent  = 0;
+    this->m_lock.lock();
```

Comment:
What is the strategy behind all the mutex calls in this file? It is to avoid using atomic member variables?

Reply:
Yes, these files were all designed to use `mutex` instead of `atomic`.  I am just improving the locking of these things, but not switching to atomic in this PR.